tests/utilities/test_tools.py

```diff
@@ -69,6 +73,66 @@ def test_asciistr(self):
             asciistr(53)
 
 
+class TestCSVWriter:
+    def test_csv_writer(self, tmp_path):
+        # Some dummy data to write to file
+        x_vals = [0, 1, 2]
+        z_vals = [-1, 0, 1]
+        flux_vals = [10, 15, 20]
+        data = np.array([x_vals, z_vals, flux_vals]).T
+        header = "This is a test\nThis is a second line"
+        col_names = ["x", "z", "heat_flux"]
+
+        # Write the data to csv, using default extension and comment style
+        test_output_base = tmp_path.as_posix() + "csv_write_dummy_data"
+        write_csv(data, test_output_base, col_names, header)
+
+        # Retrieve data file to compare
+        expected_csv_output = (
+            get_bluemira_root() + "/tests/utilities/test_data/test_csv_writer.csv"
+        )
+
+        # Compare
+        test_csv_output = test_output_base + ".csv"
+
+        assert filecmp.cmp(test_csv_output, expected_csv_output)
+
+        # Clean up
+        Path(test_csv_output).unlink()
+
```

Comment:
We dont remove resultant test files because they're stored in a temporary folder that a user might want to look at post test run
```suggestion
```